cdci_data_analysis/analysis/drupal_helper.py

```diff
@@ -52,39 +55,81 @@ def execute_drupal_request(url,
-            if params is None:
-                params = {}
-            params['_format'] = request_format
-            return requests.post(url,
-                                 params={**params},
-                                 data=data,
-                                 files=files,
-                                 headers=headers
-                                 )
-    except Exception as e:
-        logger.warning(f"an issue occurred when performing a request to the product gallery, "
-                       f"this is likely to be a connection related problem, we are investigating and "
-                       f"try to solve it as soon as possible")
-        if sentry_client is not None:
-            sentry_client.capture('raven.events.Message',
-                                  message=f'exception when performing a request to the product gallery: {repr(e)}')
-        else:
-            logger.warning("sentry not used")
-
-        raise InternalError('issue when performing a request to the product gallery',
-                         status_code=500,
-                         payload={'error_message': str(e)})
+    n_tries_left = n_max_tries
+    while True:
+        try:
+            if method == 'get':
+                if params is None:
+                    params = {}
+                params['_format'] = request_format
+                res = requests.get(url,
+                                   params={**params},
+                                   headers=headers)
+
+            elif method == 'post':
+                if data is None:
+                    data = {}
+                if params is None:
+                    params = {}
+                params['_format'] = request_format
+                res = requests.post(url,
+                                    params={**params},
+                                    data=data,
+                                    files=files,
+                                    headers=headers
+                                    )
+            else:
+                raise NotImplementedError
+            if res.status_code == 403:
+                try:
+                    response_json = res.json()
+                    # a 403 has been noticed to be returned in two different cases:
+                    # * for not-valid token
+                    # * not-completed request
+                    error_msg = response_json['message']
+                except json.decoder.JSONDecodeError:
+                    error_msg = res.text
+                raise RequestNotAuthorized(error_msg)
+
+            if res.status_code in 404:
+                logger.warning(f"there seems to be some problem in completing a request to the product gallery,"
+                               " this might be due to a an error in the url, "
+                               "or because we are attempting to load a page that no longer exists, "
+                               "please check it and try to issue again the request")
```

Comment:
could you add here in the warning and also in the exception the page which was requested, and `res.text`. Would be very helpful.